rag_experiment_accelerator/run/index.py

```diff
@@ -42,68 +112,432 @@ def run(config_dir: str, data_dir: str = "data", filename: str = "config.json")
             " ensure you have the proper permissions and try again"
         )
         raise e
-    index_dict = {"indexes": []}
 
-    for chunk_size in config.CHUNK_SIZES:
-        for overlap in config.OVERLAP_SIZES:
-            for embedding_model in config.embedding_models:
-                for ef_construction in config.EF_CONSTRUCTIONS:
-                    for ef_search in config.EF_SEARCHES:
-                        index_name = get_index_name(
-                            config.NAME_PREFIX,
-                            chunk_size,
-                            overlap,
-                            embedding_model.name,
-                            ef_construction,
-                            ef_search,
-                        )
-                        logger.info(f"Creating Index with name: {index_name}")
-                        create_acs_index(
-                            service_endpoint,
-                            index_name,
-                            key,
-                            embedding_model.dimension,
-                            ef_construction,
-                            ef_search,
-                            config.LANGUAGE["analyzers"],
-                        )
-                        index_dict["indexes"].append(index_name)
 
-    index_output_file = f"{config.artifacts_dir}/generated_index_names.jsonl"
-    with open(index_output_file, "w") as index_name:
-        json.dump(index_dict, index_name, indent=4)
+def chunk_size_overlap_combinations(config: Config):
+    """
+    Generates combinations for different chunk sizes and overlaps defined
+    in the configuration
+
+    Args:
+        config (object): Configuration for rag accelerator
+
+    Returns:
+        A list of tuples for each chunk size and overlap combination
+    """
+
+    return itertools.product(config.CHUNK_SIZES, config.OVERLAP_SIZES)
+
+
+def create_search_index(config, service_endpoint, key, index_dict, chunk_size, overlap):
+    """
+    Creates a search index in Azure Search.
+
+    This function constructs the index name based on various configuration
+    parameters and then creates the index using the `create_acs_index`
+    function.
+    It logs the start and end of the index creation process and the time took.
+    The created index name is then added to the `index_dict` dictionary.
+
+    Args:
```

Comment:
These args don't appear to match the function signature.
e.g. `dimension`
Is this a CoPilot hallucination?

Reply:
I think I've removed some of the arguments while refactoring the code and forgot to update the doc string

Reply:
I think this is still an issue

Reply:
![image](https://github.com/microsoft/rag-experiment-accelerator/assets/3179107/029f3e13-da26-46e0-9fc8-e630b714ce2d)